### PR TITLE
Summation of the minisummit blogpost

### PR DESCRIPTION
**Pull request by KBanecka**

### CODE REVIEW

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+Secure Encrypted Visualisation (SEV) and Secure Memory Encryption (SME) with
+encrypted state (SEV-ES) extension and explained why SME is better than TSME.
+Afterwards, we have been introduced with status of mentioned features on Open
+Source, debating whether it is suppoorted, whether it can be used, what impact
+Qubes OS will have on the features or is it possible to run SEV-enabled guests
+using open-source tools. Michał described briefly also OVMF and SUSE Linux
+Enterprise Server support for SEV, Automatic Exits (AE) and Non-automatic Exits
+(NAE). After prelection, our specialists took the floor debating over host
+system protection, memory encryption on the BIOS level, XEN support and many
+more issues that can be found in the video above.
```
Comment (pietrushnic) [inline blog/content/post/2020-06-15-qubes_summation.md:81]: @KBanecka I think we should consider 2 things for each presentation

- how community can engage
- what we hope to do in next 12 months

I believe you can ask authors of presentations

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+overview
+  2. Marek Marczykowski-Górecki: Operating system testing, when it itself uses
+  virtual machines
+  3. AMA session
+
+### Qubes OS future: AMD SEV and fwupd/LVFS
+
+ The first day of 'minisummit' gathered us around Qubes OS future: AMD SEV and
+fwupd/LVFS, where prelegents were discussing projects based on Qubes OS. The
+opening presetation was held by Michał Żygowski, Firmware Engineer in 3mdeb,
```
Comment (pietrushnic) [inline blog/content/post/2020-06-15-qubes_summation.md:68]: @KBanecka please link also to profiles on this website

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+more issues that can be found in the video above.
+
+The second prelegent was Norbert Kamiński, Junior Embedded Systems Engineer in
+3mdeb and open-source contributor, who has introduced us status of fwupd/LVFS
+support for Qubes OS. Starting with overall informations about Firmware Update
+and Linux Vendor Firmware Service, Norbert described three layers of it’s
+architecture (system, session, internet) and explained how exactly works secure
+web service used by hardware vendors to upload firmware archives. He also
+presented fwupdmgr CLI, client tool for manual update process, describing it’s
+role between LVFS database and fwupd. Inside project status, sponsored by NLNET
```
Comment (pietrushnic) [inline blog/content/post/2020-06-15-qubes_summation.md:90]: @KBanecka NLnet Foundation

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+foundation, we’ve heard about Qubes OS support challenges and architecture
+solutions, architecture plan of the fwupd/LVFS support for Qubes OS based on
+sys-usb, dom0 and UpdateVM, process of downloading and updating firmware and
+what is going to be done in the next steps of development: qubes fwupdmgr script
+and .cab arhives validation. After the presentation came many questions around
+the topic. We were considering how .cab archives are signed, Qubes Dom 0 update
+issues, .cab file validation tool, sevices that provide proofs of
+reproducibility and many more.
+
+### Different approaches of using RTMS in Qubes OS
```
Comment (pietrushnic) [inline blog/content/post/2020-06-15-qubes_summation.md:100]: RTMs?

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+The second day of the 'minisummit' was devoted to different approaches of using
+RTMS in Qubes OS. Piotr Król, Founder and Embedded Systems Consultant, has
+introduced SRTM for Qubes OS VMS, covering feasibility and security of various
+S-RTM implementations for Qubes OS virtual machines. Piotr has presented S-RTM,
+giving practical use cases and explained how it is created on real hardware,
+presenting ways of moving it to VMs. He described Xen stub domains and explained
+how to enable TPM in QEMU S-RTM, considering options of bootfirmare,
```
Comment (pietrushnic) [inline blog/content/post/2020-06-15-qubes_summation.md:108]:
```suggestion
how to enable TPM in QEMU, considering options of boot firmware,
```

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+bootloaders, operating systems and Xen version. We were broadly familiarised
+with libtpm-based TPM emulator swtpm, Xen vTPMs and finally with assumptions and
+future ideas for 3mdeb. Among Q&A questions specialists were debating over an
+option for hosting QUEM emulation outside of QUEM and general idea of TPM inside
```
Comment (pietrushnic) [inline blog/content/post/2020-06-15-qubes_summation.md:112]:
```suggestion
option for hosting QEMU emulation outside of QEMU and a general idea of TPM inside
```

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+VM.
```
Comment (pietrushnic) [inline blog/content/post/2020-06-15-qubes_summation.md:113]:
```suggestion
VM. 3mdeb looks for possible founding of S-RTM effort either through the foundation, commercial agreement, or community effort.
```

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+
+Michał Żygowski introduced Anti Evil Maid for modern AMD UEFI-based platform,
+where he has explained what exactly Evil Maid attacs are and how can we protect
```
Comment (pietrushnic) [inline blog/content/post/2020-06-15-qubes_summation.md:116]:
```suggestion
where he has explained what exactly Evil Maid attacks are and how can we protect
```

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+ourselves from them with Qubes OS Ani-Evil-Maid. Michał presented current status
+of AEM, explained what it provides and which attacs are still not prevented.
```
Comment (pietrushnic) [inline blog/content/post/2020-06-15-qubes_summation.md:118]:
```suggestion
of AEM, explained what it provides and which attacks are still not prevented.
```

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+Intel in AEM script and more.
+
+### DRTM in Qubes OS – not only for DOM0 but also for VMS
+
+The third day was devoted to DRTM in Qubes OS – not only for DOM0 but also for
+VMS. Piotr Król has introduced DRTM for Qubes OS VMS, discussing the value and
+usage models of D-RTM implementation in Qubes OS. He started presenting Root of
+Trust family discussing D-RTM and how does it differ from S-RTM proceeding with
+an overview of boot process for Qubes OS with Early/Late Launch scenario and
+Flicker session. Based on flicker, Piotr has proposed scenarios in which D-RTM
```
Comment (pietrushnic) [inline blog/content/post/2020-06-15-qubes_summation.md:131]:
```suggestion
Flicker session. Based on Flicker, Piotr has proposed scenarios in which D-RTM
```

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+may be used: Platform Relaunch, Virtual Machine Introspection technique and
```
Comment (pietrushnic) [inline blog/content/post/2020-06-15-qubes_summation.md:132]:
```suggestion
may be used for: Platform Relaunch, Virtual Machine Introspection technique and
```

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+D-RTM, vTPM and D-RTM, Network booted vDLME and Visual Trust level indicator for
+Vms. The presentation has ended with future ideas, among others trusted system
```
Comment (pietrushnic) [inline blog/content/post/2020-06-15-qubes_summation.md:134]:
```suggestion
VMs. The presentation has ended with future ideas, among others trusted system
```

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+backups and migration, trusted firmware update, dynamic RPC policy or secure
+storage. The broad discussion covered among others the best way to re-establish
+trust in the platform.
```
Comment (pietrushnic) [inline blog/content/post/2020-06-15-qubes_summation.md:137]:
```suggestion
trust in the platform. D-RTM is a vast and complicated topic and requires much more education and development. Our work related to [OpenDRTM for AMD using TrenchBoot](https://nlnet.nl/project/OpenDRTM/) founded by NLnet should move the ecosystem forward, but we still looking at how to advance Intel-based solutions. There is also quite a lot of work in Xen and Linux kernel. We believe that Qubes OS minisummit 2021 should cover practical demos of TrenchBoot as reference open source D-RTM implementation.
```

**Comment by pietrushnic** [comment]
@KBanecka spellchecker would be appreciated before pushing for review

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 14]
```diff
@@ -0,0 +1,184 @@
...
+
+---
+The second Qubes OS and 3mdeb 'minisummit' is ahead of us. We have gone through
+four evenings of topics devoted to Qubes OS. It has been another fruitful and
+productive event, during which we have exchanged loads of knowledge and
+expertise. Below you will find list of topics, we have discussed with related
+videos and broad report of each day.
+
+
+* 20.05
```
Comment (miczyg1) [inline blog/content/post/2020-06-15-qubes_summation.md:34]: @KBanecka please also add slides for my presentations: https://cloud.3mdeb.com/index.php/f/247810
There are 3 PDFs there. For example:

```
  1. Michał Żygowski: Qubes OS on modern AMD platform
    * [recording](https://www.youtube.com/watch?v=Rw7rAPPyPPc&t=31s)
    * [slides](link)
```

@pietrushnic @Asiderr maybe consider publishing slides?

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 15]
```diff
@@ -0,0 +1,184 @@
...
+
+### Qubes OS future: AMD SEV and fwupd/LVFS
+
+ The first day of 'minisummit' gathered us around Qubes OS future: AMD SEV and
+fwupd/LVFS, where prelegents were discussing projects based on Qubes OS. The
+opening presetation was held by Michał Żygowski, Firmware Engineer in 3mdeb,
+Braswell SoC, PC Engines and Protectli maintainer in coreboot who has presented
+Qubes on SuperMicro M11SDV-4C-LN4F superserver platform, that has been tested
+with Qubes OS on board. Michał has introduced and explained it’s features,
+Secure Encrypted Visualisation (SEV) and Secure Memory Encryption (SME) with
```
Comment (miczyg1) [inline blog/content/post/2020-06-15-qubes_summation.md:72]:
```suggestion
Secure Memory Encryption (SME) and Secure Encrypted Visualisation (SEV) with
```

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 16]
```diff
@@ -0,0 +1,184 @@
...
+and Linux Vendor Firmware Service, Norbert described three layers of it’s
+architecture (system, session, internet) and explained how exactly works secure
+web service used by hardware vendors to upload firmware archives. He also
+presented fwupdmgr CLI, client tool for manual update process, describing it’s
+role between LVFS database and fwupd. Inside project status, sponsored by NLNET
+foundation, we’ve heard about Qubes OS support challenges and architecture
+solutions, architecture plan of the fwupd/LVFS support for Qubes OS based on
+sys-usb, dom0 and UpdateVM, process of downloading and updating firmware and
+what is going to be done in the next steps of development: qubes fwupdmgr script
+and .cab arhives validation. After the presentation came many questions around
```
Comment (miczyg1) [inline blog/content/post/2020-06-15-qubes_summation.md:95]:
```suggestion
and .cab archives validation. After the presentation many questions came around
```

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 4 above)
Comment (miczyg1) [reply blog/content/post/2020-06-15-qubes_summation.md:100]: Root of Trust for Measurement = RTM, so it should be RTMs @KBanecka

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 17]
```diff
@@ -0,0 +1,184 @@
...
+the topic. We were considering how .cab archives are signed, Qubes Dom 0 update
+issues, .cab file validation tool, sevices that provide proofs of
+reproducibility and many more.
+
+### Different approaches of using RTMS in Qubes OS
+
+The second day of the 'minisummit' was devoted to different approaches of using
+RTMS in Qubes OS. Piotr Król, Founder and Embedded Systems Consultant, has
```
Comment (miczyg1) [inline blog/content/post/2020-06-15-qubes_summation.md:103]: a/a

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 18]
```diff
@@ -0,0 +1,184 @@
...
+how to enable TPM in QEMU S-RTM, considering options of bootfirmare,
+bootloaders, operating systems and Xen version. We were broadly familiarised
+with libtpm-based TPM emulator swtpm, Xen vTPMs and finally with assumptions and
+future ideas for 3mdeb. Among Q&A questions specialists were debating over an
+option for hosting QUEM emulation outside of QUEM and general idea of TPM inside
+VM.
+
+Michał Żygowski introduced Anti Evil Maid for modern AMD UEFI-based platform,
+where he has explained what exactly Evil Maid attacs are and how can we protect
+ourselves from them with Qubes OS Ani-Evil-Maid. Michał presented current status
```
Comment (miczyg1) [inline blog/content/post/2020-06-15-qubes_summation.md:117]:
```suggestion
ourselves from them with Qubes OS Anti-Evil-Maid. Michał presented the current status
```

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 19]
```diff
@@ -0,0 +1,184 @@
...
+of AEM, explained what it provides and which attacs are still not prevented.
+Afterwards, prelegent described ways of enabling AEM in Qubes OS for AMD
+platform and for TPM2 showing instalation steps, sources and repositories.
+During Q&A we were considering how does it work with UEFI, what differs AMD from
+Intel in AEM script and more.
+
+### DRTM in Qubes OS – not only for DOM0 but also for VMS
+
+The third day was devoted to DRTM in Qubes OS – not only for DOM0 but also for
```
Comment (miczyg1) [inline blog/content/post/2020-06-15-qubes_summation.md:126]:
```suggestion
The third day was devoted to DRTM in Qubes OS – not only for Dom0 but also for
```

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 20]
```diff
@@ -0,0 +1,184 @@
...
+VMS. Piotr Król has introduced DRTM for Qubes OS VMS, discussing the value and
```
Comment (miczyg1) [inline blog/content/post/2020-06-15-qubes_summation.md:127]:
```suggestion
VMs. Piotr Król has introduced DRTM for Qubes OS VMs, discussing the value and
```

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 21]
```diff
@@ -0,0 +1,184 @@
...
+fuzzing tests, build tests, integration tests, installation tests, presented
+openQA tool used for running tests in VM and described missing parts of testing
+Qubes. During Q&A session we were considering how the community can help
+extending test scope and what kind of test are the most necessarry for Qubes OS
+development.
+
+
+## Summary
+
+ Qubes OS & 3mdeb 'minisummit' has been a great event that triggered many
```
Comment (miczyg1) [inline blog/content/post/2020-06-15-qubes_summation.md:169]: Obsolete space?

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 22]
```diff
@@ -0,0 +1,184 @@
...
+important questions and allowed the community to take an active part in the
+broad discussion around covered issues. We would like to thank Marek and
+Frédéric for sharing your knowledge and time through all the events. Thank you
+for your cooperation and willingness to ask and answer. Great appreciation goes
+towards all the community that joined our meetings and many times convinced us
+that events such as 'minisummit' are important and necessary. We have exchanged
+loads of knowledge and experience and we would like to meet again, on Qubes OS
+and 3mdeb 'minisummit' 2021. Thank you all once again.
+
+ If you think we can help in improving Qubes OS support for your hardware, help
```
Comment (miczyg1) [inline blog/content/post/2020-06-15-qubes_summation.md:179]: a/a

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 23]
```diff
@@ -0,0 +1,184 @@
...
+    * [recording](https://youtu.be/rM0vRi6qABE?t=1904)
+
+* 04.06
+  1. Piotr Król: DRTM for Qubes OS VMs
+    * [recording](https://youtu.be/pZF-jyJWTE4)
+  2. Michał Żygowski: Anti Evil Maid for Intel coreboot-based platform
+    * [recording](https://youtu.be/YE2FbFlszI4?t=9)
+  3. AMA session
+    * [recording](https://youtu.be/YE2FbFlszI4?t=1725)
+* 10.06
```
Comment (artur-rs) [inline blog/content/post/2020-06-15-qubes_summation.md:57]: @KBanecka do we have recordings already?

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 2 above)
Comment (artur-rs) [inline blog/content/post/2020-06-15-qubes_summation.md:68]: @KBanecka `presentation`

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 24]
```diff
@@ -0,0 +1,184 @@
...
+ The first day of 'minisummit' gathered us around Qubes OS future: AMD SEV and
+fwupd/LVFS, where prelegents were discussing projects based on Qubes OS. The
+opening presetation was held by Michał Żygowski, Firmware Engineer in 3mdeb,
+Braswell SoC, PC Engines and Protectli maintainer in coreboot who has presented
+Qubes on SuperMicro M11SDV-4C-LN4F superserver platform, that has been tested
+with Qubes OS on board. Michał has introduced and explained it’s features,
+Secure Encrypted Visualisation (SEV) and Secure Memory Encryption (SME) with
+encrypted state (SEV-ES) extension and explained why SME is better than TSME.
+Afterwards, we have been introduced with status of mentioned features on Open
+Source, debating whether it is suppoorted, whether it can be used, what impact
```
Comment (artur-rs) [inline blog/content/post/2020-06-15-qubes_summation.md:75]: @KBanecka `supported`

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 16 above)
Comment (artur-rs) [inline blog/content/post/2020-06-15-qubes_summation.md:95]: @KBanecka maybe wrap each .cab (\`.cab\`) to get code syntax: `.cab`

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 25]
```diff
@@ -0,0 +1,184 @@
...
+develop, on the example of Qubes builder v.1 and v.2.
+
+The last prelegent, Marek Marczykowski-Górecki, Project lead in QubesOS has
+introduced operating system testing, when it itself uses virtual machines. Marek
+briefly presented how to test Qubes OS, by describing python unit tests and
+pytest frameworks, used inside Qubes OS with Travis CI. He has also mentioned
+fuzzing tests, build tests, integration tests, installation tests, presented
+openQA tool used for running tests in VM and described missing parts of testing
+Qubes. During Q&A session we were considering how the community can help
+extending test scope and what kind of test are the most necessarry for Qubes OS
```
Comment (artur-rs) [inline blog/content/post/2020-06-15-qubes_summation.md:163]: @KBanecka `necessary`

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 14 above)
Comment (pietrushnic) [reply blog/content/post/2020-06-15-qubes_summation.md:34]: https://www.slideshare.net/PiotrKrl/srtm-for-qubes-os-vms
https://www.slideshare.net/PiotrKrl/drtm-for-qubes-os-vms

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 23 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md:60]: I've added recordings.

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 14 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md:36]: I am waiting for Norbert and Marek presentations

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 22 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md:208]: I would rather leave this space here

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 2 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md [hunk 26]
```diff
@@ -0,0 +1,184 @@
+---
+title: Qubes OS & 3mdeb 'minisummit' 2020 summation
+abstract: 'The second Qubes OS & 3mdeb minisummit is ahead of us. We had gone through four evenings of topics devoted to Qubes OS, so it is time for broad summation of the event.'
```
[the remark this answers is not in the document]
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 3 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 4 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 9 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 10 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 11 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 12 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 13 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 15 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 16 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 17 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 18 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 19 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 20 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 21 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: deleted

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 2 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: corrected

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 24 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: corrected

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 16 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 25 above)
Comment (KBanecka) [reply blog/content/post/2020-06-15-qubes_summation.md]: done

File: blog/content/post/2020-06-15-qubes_summation.md (hunk 22 above)
Comment (miczyg1) [reply blog/content/post/2020-06-15-qubes_summation.md:208]: Why? I don't recall any blog post having the space here

**Pull request merged**